src/main/java/org/takes/facets/auth/codecs/CcStrict.java

```diff
@@ -67,7 +67,8 @@ private static Identity applyRules(final Identity identity) {
         if (identity.urn().isEmpty()) {
             throw new DecodingException("urn is empty");
         }
-        if (!identity.urn().matches("urn:[a-z]+:.+")) {
+        // @checkstyle LineLength (1 line)
+        if (!identity.urn().matches("^(?i)^urn(?-i):[a-z]{1,31}(:([\\-a-zA-Z0-9/]|%[0-9a-fA-F]{2})*)+(\\?\\w+(=([\\-a-zA-Z0-9/]|%[0-9a-fA-F]{2})*)?(&\\w+(=([\\-a-zA-Z0-9/]|%[0-9a-fA-F]{2})*)?)*)?\\*?$")) {
```

Comment:
@dmzaytsev Wouldn't it be better to add a dependency to the `jcabi-urn` project and use method `boolean URN.isValid(String text)`?
@yegor256 what do you think?